pkg/server/server.go

```diff
@@ -377,35 +378,51 @@ func createDeployment(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	//TODO: Duplicated code, could be moved to helper function
-	//Get JSON from url
-	httpClient := &http.Client{}
+	//Needs to be at higher scope than if statement
+	tempPTS := &api.PodTemplateSpec{}
+	//Check if we got a URL or a direct PTS
+	if tempJSON.PTS == nil {
+		//No PTS so check ptsURL
+		fmt.Printf("No PTS\n")
+		if tempJSON.PtsURL == "" {
+			//No URL either so error
+			http.Error(w, "", http.StatusInternalServerError)
+			fmt.Printf("No ptsURL or PTS given\n")
+			return
+		}
+		//Get from URL
+		//TODO: Duplicated code, could be moved to helper function
+		//Get JSON from url
+		httpClient := &http.Client{}
 
-	req, err := http.NewRequest("GET", tempJSON.PtsURL, nil)
-	req.Header.Add("Content-Type", "application/json")
+		req, err := http.NewRequest("GET", tempJSON.PtsURL, nil)
+		req.Header.Add("Content-Type", "application/json")
 
-	//TODO: In the future if we require a secret to access the PTS store
-	// then this call will need to pass in that key.
-	urlJSON, err := httpClient.Do(req)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		fmt.Printf("Error retrieving pod template spec: %v\n", err)
-		return
-	}
-	defer urlJSON.Body.Close()
+		//TODO: In the future if we require a secret to access the PTS store
+		// then this call will need to pass in that key.
+		urlJSON, err := httpClient.Do(req)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			fmt.Printf("Error retrieving pod template spec: %v\n", err)
+			return
+		}
+		defer urlJSON.Body.Close()
 
-	if urlJSON.StatusCode != 200 {
-		fmt.Printf("Expected 200 got: %v\n", urlJSON.StatusCode)
-		http.Error(w, "", http.StatusInternalServerError)
-		return
-	}
+		if urlJSON.StatusCode != 200 {
+			fmt.Printf("Expected 200 got: %v\n", urlJSON.StatusCode)
+			http.Error(w, "", http.StatusInternalServerError)
+			return
+		}
 
-	tempPTS := &api.PodTemplateSpec{}
-	err = json.NewDecoder(urlJSON.Body).Decode(tempPTS)
-	if err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		fmt.Printf("Error decoding PTS JSON Body: %v\n", err)
-		return
+		err = json.NewDecoder(urlJSON.Body).Decode(tempPTS)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			fmt.Printf("Error decoding PTS JSON Body: %v\n", err)
+			return
+		}
+	} else {
+		//We got a direct PTS so just copy it
+		tempPTS = tempJSON.PTS
```

Comment:
Yes it will be validated when it is decoded. Additionally it will be checked by the kubernetes apiserver.